test/unit/libraries/ReservoirLibrary.t.sol

```diff
@@ -8,6 +8,84 @@ import { ReservoirLibrary } from "src/libraries/ReservoirLibrary.sol";
 
 contract ReservoirLibraryTest is BaseTest
 {
+    function testGetSwapFee() public
+    {
+        // assert
+        assertEq(ReservoirLibrary.getSwapFee(address(_factory), address(_tokenA), address(_tokenB), 0), 3000);
+    }
+
+    // commented out as vm.expectRevert does not work properly on library functions
+//    function testGetSwapFee_PairDoesNotExist() public
+//    {
+//        // act & assert
+//        vm.expectRevert();
+//        ReservoirLibrary.getSwapFee(address(_factory), address(_tokenA), address(_tokenC), 0);
+//    }
+
+    function testGetPrecisionMultiplier() public
+    {
+        // arrange
+        MintableERC20 l0DecimalToken = new MintableERC20("zero", "0", 0);
+
+        // act & assert
+        assertEq(ReservoirLibrary.getPrecisionMultiplier(address(_tokenA)), 1);
+        assertEq(ReservoirLibrary.getPrecisionMultiplier(address(_tokenD)), 1e12);
+        assertEq(ReservoirLibrary.getPrecisionMultiplier(address(l0DecimalToken)), 1e18);
+    }
+
+    // commented out as vm.expectRevert does not work properly on library functions
```

Comment:
`vm.expectRevert` operates on `CALL`, a library function is not a `CALL` if it is inlined (which `internal` functions are). This will never work, your best bet is to create a dummy class that performs the call internally. Then you can `CALL` the dummy class and observe the result of the inline call from your test contract. e.g. `TestContract -> Dummy --internal-> Library`

Reply:
yup I understand the mechanics of this, as you explained to me previously. 

Currently for some calling of lib functions, `expectRevert` will work, sometimes it won't. Foundry [wants to fix this](https://github.com/foundry-rs/foundry/issues/3437) by making it `expectRevert` always fail for internal calls. 

I'll make a dummy contract to use it

Reply:
fixed